wasm/errors.go

```diff
@@ -30,13 +30,23 @@ const (
 	ErrorCodeUnknownCommandRequest
 	ErrorCodeUnknownCommandResponse
 	ErrorCodeMemoryOutOfRange
+
+	ErrorCodeSchemaUnmarshal
+	ErrorCodeSchemaMarshal
+	ErrorCodeSchemaGet
+	ErrorCodeSchemaCreate
 )
 
 var (
 	ErrNoMoreCommands         = NewError(ErrorCodeNoMoreCommands, "no more commands")
 	ErrUnknownCommandRequest  = NewError(ErrorCodeUnknownCommandRequest, "unknown command request")
 	ErrUnknownCommandResponse = NewError(ErrorCodeUnknownCommandResponse, "unknown command response")
 	ErrMemoryOutOfRange       = NewError(ErrorCodeMemoryOutOfRange, "memory out of range")
+
+	ErrSchemaUnmarshal = NewError(ErrorCodeSchemaUnmarshal, "failed to unmarshal the proto request")
+	ErrSchemaMarshal   = NewError(ErrorCodeSchemaMarshal, "failed to marshal the proto response into a schema")
+	ErrSchemaGet       = NewError(ErrorCodeSchemaGet, "error while getting the schema")
+	ErrSchemaCreate    = NewError(ErrorCodeSchemaCreate, "error while creating the schema")
```

Comment:
These are pretty generic errors, I would think about changing them into general-purpose errors that can be reused in any host call. For instance, the umarshalling/marshalling error can happen in any host call. Regarding schema errors, we need more specific ones like "schema not found" when getting a schema or "invalid schema" when creating it. Maybe there's more.

I think we also need to provide equivalent error structs in the root of the repo so the processor can use them. The processor code shouldn't directly use this code, but it needs to be able to know if a schema was not found for instance.

We can pair on this if you want to figure out the proper way to structure this.